haystack_experimental/components/generators/chat/openai.py

```diff
@@ -0,0 +1,155 @@
+        generation_kwargs: Optional[dict[str, Any]] = None,
+        *,
+        tools: Optional[Union[list[Tool], Toolset]] = None,
+        tools_strict: Optional[bool] = None,
+        hallucination_score_config: Optional[HallucinationScoreConfig] = None,
+    ) -> dict[str, list[ChatMessage]]:
+        """
+        Asynchronously invokes chat completion based on the provided messages and generation parameters.
+
+        This is the asynchronous version of the `run` method. It has the same parameters and return values
+        but can be used with `await` in async code.
+
+        :param messages:
+            A list of ChatMessage instances representing the input messages.
+        :param streaming_callback:
+            A callback function that is called when a new token is received from the stream.
+            Must be a coroutine.
+        :param generation_kwargs:
+            Additional keyword arguments for text generation. These parameters will
+            override the parameters passed during component initialization.
+            For details on OpenAI API parameters, see [OpenAI documentation](https://platform.openai.com/docs/api-reference/chat/create).
+        :param tools:
+            A list of tools or a Toolset for which the model can prepare calls. If set, it will override the
+            `tools` parameter set during component initialization. This parameter can accept either a list of
+            `Tool` objects or a `Toolset` instance.
+        :param tools_strict:
+            Whether to enable strict schema adherence for tool calls. If set to `True`, the model will follow exactly
+            the schema provided in the `parameters` field of the tool definition, but this may increase latency.
+            If set, it will override the `tools_strict` parameter set during component initialization.
+        :param hallucination_score_config:
+            If provided, the generator will evaluate the hallucination risk of its responses using
+            the OpenAIPlanner and annotate each response with hallucination metrics.
+            This involves generating multiple samples and analyzing their consistency, which may increase
+            latency and cost. Use this option when you need to assess the reliability of the generated content
+            in scenarios where accuracy is critical.
+
+        :returns:
+            A dictionary with the following key:
+            - `replies`: A list containing the generated responses as ChatMessage instances. If hallucination
+              scoring is enabled, each message will include additional metadata:
+                - `hallucination_decision`: "ANSWER" if the model decided to answer, "REFUSE" if it abstained.
+                - `hallucination_risk`: The EDFL hallucination risk bound.
+                - `hallucination_rationale`: The rationale behind the hallucination decision.
+        """
+        if len(messages) == 0:
+            return {"replies": []}
+
```

Comment:
We could do some validation of the HallucinationScoreConfig here but given that this is in haystack-experimental, I agree that we don't need it here.